KLSadd_insertion/src/updateChapters.py

```diff
@@ -164,21 +390,19 @@ def referencePlacer(chap, references, p, kls, refII, refIII, specref):
         designator = "9."
     elif chapticker2 == 1:
         designator = "14."
+
     for i in references:
         # Place before References paragraph
         word1 = str(p[count])
-        if (designator in word1[word1.find(
-                "\\subsection*{") + 1: word1.find("}")]):
+        if designator in word1[word1.find("\\subsection*{") + 1: word1.find("}")]:
             chap[i - 2] += "%Begin KLSadd additions"
             chap[i - 2] += p[count]
             chap[i - 2] += "%End of KLSadd additions"
             count += 1
         else:
```

Comment:
The `else` isn't actually necessary. The condition of the `if` and the condition of the `while` are mutually exclusive, so the code will never enter the `while` after entering the `if` anyway. Also, consider either making the `if` statement a function (as it is repeated directly after the `while` header) or restructuring the code so that there isn't the repetition of the statement.